edge/pkg/metamanager/metaserver/server.go

```diff
@@ -266,3 +218,97 @@ func BuildHandlerChain(handler http.Handler, ls *MetaServer) http.Handler {
 	handler = genericfilters.WithPanicRecovery(handler, &apirequest.RequestInfoFactory{})
 	return handler
 }
+
+func (ls *MetaServer) prepareServer() {
+	err := setupDummyInterface()
+	if err != nil {
+		panic(fmt.Errorf("setup dummy interface err: %v", err))
+	}
+	certIPs, err := ls.getCertIPs()
+	if err != nil {
+		panic(fmt.Errorf("failed to get cert IP: %v", err))
+	}
+
+	certificateManager, err := certificate.NewServerCertificateManager(
+		certificate.NewSimpleClientset(),
+		types.NodeName(metaserverconfig.Config.NodeName),
+		certIPs,
+		certificate.CertificatesDir)
+	if err != nil {
+		panic(fmt.Errorf("failed to initialize certificate manager: %v", err))
+	}
+
+	err = certificateManager.WaitForCAReady()
+	if err != nil {
+		panic(fmt.Errorf("wait for CA ready failed: %v", err))
+	}
+
+	certificateManager.Start()
+	err = certificateManager.WaitForCertReady()
+	if err != nil {
+		panic(fmt.Errorf("wait for cert ready failed: %v", err))
+	}
+
+	ls.serverCeriticateManager = certificateManager
+}
+
+func (ls *MetaServer) getCertIPs() ([]net.IP, error) {
+	ip, _, err := net.SplitHostPort(metaserverconfig.Config.Server)
+	if err != nil {
+		return nil, err
+	}
+	dummyIP, _, err := net.SplitHostPort(metaserverconfig.Config.DummyServer)
+	if err != nil {
+		return nil, err
+	}
+	return []net.IP{net.ParseIP(ip), net.ParseIP(dummyIP)}, nil
+}
+
+func (ls *MetaServer) makeTLSConfig() (*tls.Config, error) {
+	ca, err := os.ReadFile(fmt.Sprintf("%s/ca.crt", certificate.CertificatesDir))
+	if err != nil {
+		return nil, fmt.Errorf("read CA failed: %v", err)
+	}
+
+	block, _ := pem.Decode(ca)
+	pool := x509.NewCertPool()
+	if !pool.AppendCertsFromPEM(pem.EncodeToMemory(&pem.Block{Type: certutil.CertificateBlockType, Bytes: block.Bytes})) {
+		return nil, fmt.Errorf("failed to load ca content")
```

Comment:
You can use the function `errors.New()` when without formatting.

Reply:
> You can use the function `errors.New()` when without formatting.

Done.